### PR TITLE
Adding the option of custom logs (e.g. learning rate) to tensorboard callback

### PR DESCRIPTION
**Pull request by jundl77**
**Problem**: Right now TensorBoard only tracks ```loss```,  ```acc```, ```val_loss```, ```val_acc``` but nothing more.

I have not found a way to add custom values (e.g. the learning rate) to TensorBoard, so that they can be tracked as well. 

**Solution:** A little callback to the [TensorBoard callback](https://github.com/keras-team/keras/blob/master/keras/callbacks.py#L587) that is called at the end of each epoch, right before the logs are added to TensorBoard. It provides a way to add your custom values, so that they also show up on TensorBoard.

**Example:**

In my model:

```python
# Define tbCallback as usual, but add custom_log_func
tbCallback = TensorBoard(log_dir='./logs'
                         ...
                         custom_log_func=self.custom_log_func)
...

def custom_log_func(self, tensorboard, epoch, logs=None):
    """Add the custom logs to tensorboard. This function is executed at the end of each epoch."""

    # Add learning rate
    optimizer = tensorboard.model.optimizer
    learning_rate = ... # Calculate learning rate per epoch here

    return {
        "learning_rate": learning_rate
    }
```

The returned dict from ```custom_log_func``` is merged with the existing ```logs``` dict, so that my custom values (```learning_rate``` here) are submitted to TensorBoard as well.

### CODE REVIEW

**Comment by fchollet** [comment]
I would rather not change add more stuff to the API to cover niche use cases which can in fact be handled via simple subclassing:

```python
class XTensorBoard(TensorBoard):

    def on_epoch_end(self, epoch, logs=None):
         logs.update(my_logs_dict)
         super(XTensorBoard, self).on_epoch_end(epoch, logs)
```

In general, Keras callbacks are meant to be used via subclassing. This makes them reasonably flexible. Never hesitate to roll out your own callbacks!

**Comment by jundl77** [comment]
Ah I see, well thanks! :)

**Comment by ahundt** [comment]
I actually have this exact problem too. I guess I'll subclass, though there is something to be said for having "batteries included" when it is to allow a common task like including the learning rate.

**Comment by HuangBo-Terraloupe** [comment]
Sorry, I still do not understand how to use it. 
Can some one give more explanation or a example how to do it?
